client/js/s3/request-signer.js

```diff
@@ -153,10 +154,51 @@ qq.s3.RequestSigner = function(o) {
 
             getEncodedHashedPayload: function(body) {
                 var promise = new qq.Promise(),
-                    reader;
+                    reader,
+                    self = this;
 
                 if (qq.isBlob(body)) {
-                    // TODO hash blob in webworker if this becomes a notable perf issue
+                    // Only if a worker is requested will we load it.
+                    // We will fallback to the inline reader if the worker
+                    // fails to load.
+                    if (options.signatureSpec.useWorker) {
+                        if (!this._worker) {
```

Comment:
This file is already quite large. Ideally, this logic would be split out into another JS file.